streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -277,6 +276,24 @@ private boolean setState(final State newState) {
         return true;
     }
 
+    private boolean setRunningFromCreated() {
+        synchronized (stateLock) {
+            if (state != State.CREATED) {
+                log.error("{} Unexpected state transition from {} to {}", logPrefix, state, State.RUNNING);
+                throw new IllegalStateException(logPrefix + " Unexpected state transition from " + state + " to " + State.RUNNING);
+            }
+            state = State.RUNNING;
+            stateLock.notifyAll();
+        }
+
+        // we need to call the user customized state listener outside the state lock to avoid potential deadlocks
+        if (stateListener != null) {
+            stateListener.onChange(State.RUNNING, State.CREATED);
+        }
+
+        return true;
```

Comment:
There is only one value returned.
This method can be declared as void.

Reply:
Ack.